### PR TITLE
lrzip: enable asm on x86 and fix build on darwin

### PR DESCRIPTION
**Pull request by smancill**
###### Motivation for this change

ZHF: #144627  (@NixOS/nixos-release-managers)

###### Things done

Patch a Makefile to fix the build on darwin.

While on it, I re-enabled compilation of native assembly code on x86 instead of keeping it disabled for all systems.

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [X] x86_64-darwin
  - [ ] aarch64-darwin
- [X] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by risicle** [comment]
`nixpkgs-review` happy, macos 10.15 & nixos x86_64 & aarch64 (emulated)

**Review by risicle — APPROVED** [review]

File: pkgs/tools/compression/lrzip/default.nix [hunk 1]
```diff
@@ -1,17 +1,33 @@
-{ lib, stdenv, fetchurl, zlib, lzo, bzip2, lz4, nasm, perl }:
+{ lib, stdenv, fetchFromGitHub, autoreconfHook, zlib, lzo, bzip2, lz4, nasm, perl }:
 
+let
+  isx86 = stdenv.isx86_64 || stdenv.isi686;
```
Comment (jonringer) [inline pkgs/tools/compression/lrzip/default.nix:4]:
```suggestion
  inherit (stdenv.hostPlatform) isx86;
```

Comment (smancill) [reply pkgs/tools/compression/lrzip/default.nix]: Thanks. Updated.

There are a few other places with the same pattern:

``` sh
$ rg 'stdenv\.(hostPlatform\.)?isx86_64.*stdenv\.(hostPlatform\.)?isi686'
pkgs/os-specific/linux/firmware/fwupd/default.nix
61:  isx86 = stdenv.isx86_64 || stdenv.isi686;

pkgs/development/libraries/x264/default.nix
40:  nativeBuildInputs = lib.optional (stdenv.hostPlatform.isx86_64 || stdenv.hostPlatform.isi686) nasm;

pkgs/development/libraries/glibc/common.nix
164:    ] ++ lib.optionals (stdenv.hostPlatform.isx86_64 || stdenv.hostPlatform.isi686 || stdenv.hostPlatform.isAarch64) [

pkgs/desktops/gnome/core/gnome-software/default.nix
40:  withFwupd = stdenv.isx86_64 || stdenv.isi686;

pkgs/tools/networking/fastd/default.nix
45:  mesonFlags = lib.optionals (!stdenv.isx86_64 && !stdenv.isi686) [
```

Comment (risicle) [reply pkgs/tools/compression/lrzip/default.nix]: I think `isx86` is not present under `stdenv` at the top-level, and so many people don't know it exists.

Comment (r-burns) [reply pkgs/tools/compression/lrzip/default.nix]: I opened a PR to hopefully address this here: https://github.com/NixOS/nixpkgs/pull/146829

**Pull request merged**